src/azure-cli/azure/cli/command_modules/cdn/commands.py

```diff
@@ -207,3 +285,117 @@ def load_command_table(self, _):
         g.custom_command('delete', 'delete_waf_rate_limit_rule', client_factory=cf_waf_policy, confirmation=True)
         g.custom_command('list', 'list_waf_rate_limit_rules', client_factory=cf_waf_policy)
         g.custom_show_command('show', 'show_waf_rate_limit_rule', client_factory=cf_waf_policy)
+
+    with self.command_group('afd profile', cdn_profiles_sdk,
+                            is_preview=True,
+                            custom_command_type=get_custom_sdk(cf_profiles, _not_found(profile_not_found_msg))) as g:
+        g.custom_show_command('show', 'get_afd_profile')
+        g.custom_command('delete', 'delete_afd_profile')
+        g.custom_command('update', 'update_afd_profile')
+        g.custom_command('list', 'list_afd_profiles')
+        g.custom_command('create', 'create_afd_profile')
+        g.custom_command('usage', 'list_resource_usage', client_factory=cf_afd_profiles)
+
+    with self.command_group('afd endpoint', cdn_afd_endpoints_sdk, is_preview=True) as g:
+        g.show_command('show', 'get')
+        g.command('list', 'list_by_profile')
+        g.command('purge', 'purge_content', supports_no_wait=True)
+        g.command('delete', 'delete', confirmation=True)
+
+        g.custom_command('update', 'update_afd_endpoint', client_factory=cf_afd_endpoints)
```

Comment:
Unnecessary to add `client_factory=cf_afd_endpoints` for g.custom_command as `cdn_afd_endpoints_sdk` in self.command_group already defines client_factory as `cf_afd_endpoints`; other commands as well.

Reply:
Thanks, I will move the client_factory parameter to command group level to remove the duplication according to :
![image](https://user-images.githubusercontent.com/61817681/108961542-768adb80-76b2-11eb-862a-e8fe3419e5e9.png)

as my custom command use the client argument.

Reply:
Unnecessary to add `client_factory=cf_afd_endpoints` again cdn_afd_endpoints_sdk already defines client_factory `cf_afd_endpoints`

---
In reply to: [581681124](https://github.com/Azure/azure-cli/pull/16951#discussion_r581681124) [](ancestors = 581681124)